crates/utils/src/tracing/grpc.rs

```diff
@@ -41,120 +46,59 @@ pub fn traced_span_fn<T>(component: TracedComponent) -> fn(&http::Request<T>) ->
     }
 }
 
-fn rpc_trace_fn<T>(request: &http::Request<T>) -> tracing::Span {
-    let span = match request.uri().path().rsplit('/').next() {
-        Some("CheckNullifiers") => rpc_span!("rpc.rpc", "CheckNullifiers"),
-        Some("SyncNullifiers") => rpc_span!("rpc.rpc", "SyncNullifiers"),
-        Some("GetBlockHeaderByNumber") => rpc_span!("rpc.rpc", "GetBlockHeaderByNumber"),
-        Some("SyncStorageMaps") => rpc_span!("rpc.rpc", "SyncStorageMaps"),
-        Some("SyncAccountVault") => rpc_span!("rpc.rpc", "SyncAccountVault"),
-        Some("SyncState") => rpc_span!("rpc.rpc", "SyncState"),
-        Some("SyncNotes") => rpc_span!("rpc.rpc", "SyncNotes"),
-        Some("GetNotesById") => rpc_span!("rpc.rpc", "GetNotesById"),
-        Some("SubmitProvenTransaction") => rpc_span!("rpc.rpc", "SubmitProvenTransaction"),
-        Some("GetAccountDetails") => rpc_span!("rpc.rpc", "GetAccountDetails"),
-        Some("GetBlockByNumber") => rpc_span!("rpc.rpc", "GetBlockByNumber"),
-        Some("GetAccountProofs") => rpc_span!("rpc.rpc", "GetAccountProofs"),
-        Some("GetNoteScriptByRoot") => rpc_span!("rpc.rpc", "GetNoteScriptByRoot"),
-        Some("Status") => rpc_span!("rpc.rpc", "Status"),
-        _ => rpc_span!("rpc.rpc", "Unknown"),
-    };
+/// Creates a dynamic tracing span based on the HTTP path of the gRPC request.
+///
+/// This function extracts the method name from the HTTP path and creates a span with
+/// the service name from the component. The actual span name used by OpenTelemetry
+/// is set dynamically via the `otel.name` field, allowing us to avoid hardcoded
+/// method name matches.
+fn dynamic_trace_fn<T>(request: &http::Request<T>, component: TracedComponent) -> tracing::Span {
+    let service_name = component.service_name();
+
+    // Extract the method name from the HTTP path (e.g., "/rpc.rpc/CheckNullifiers" ->
+    // "CheckNullifiers")
+    let method_name = request.uri().path().rsplit('/').next().unwrap_or("Unknown");
```

Comment:
Now that we are not doing an exhaustive match on the method, maybe we can just unwrap_or_default() and keep it empty?